src/mds/MDCache.h

```diff
@@ -539,10 +540,16 @@ class MDCache {
     cap_imports[ino][client][frommds] = icr;
   }
   const cap_reconnect_t *get_replay_cap_reconnect(inodeno_t ino, client_t client) {
-    if (cap_imports.count(ino) &&
-	cap_imports[ino].count(client) &&
-	cap_imports[ino][client].count(MDS_RANK_NONE)) {
-      return &cap_imports[ino][client][MDS_RANK_NONE];
+    auto cap_imports_ino = cap_imports.find(ino);
+    if(cap_imports_ino != cap_imports.end()) {
+      auto cap_imports_client = cap_imports_ino->second.find(client);
+      if(cap_imports_client != cap_imports_ino->second.end())
+      {
+	auto cap_imports_frommds = cap_imports_client->second.find(MDS_RANK_NONE);
+	if(cap_imports_frommds != cap_imports_client->second.end()) {
+	  return &cap_imports_frommds->second;
+	}
+      }
```

Comment:
This does reduce the number of ops but is *way* harder to read. If we're interested in making it faster (and this is not really a fast-path thing where we care) we need to keep it easy to read. Probably making the ```if (a && b && c) ``` but expanding the conditions to assign variables and check them at the same time, instead of nesting if statements.